services/workflows-service/prisma/schema.prisma

```diff
@@ -759,8 +746,7 @@ enum SanctionListMatchStatus {
 model Counterparty {
   id String @id @default(cuid())
 
-  correlationId String?
-  type          CounterpartyType
+  correlationId String? // This column is a denormalized value of the EndUser.correlationId or Business.correlationId
```

Comment:
Kept this column as a denormalization of EndUser.correlationId or Business.correlationId. Not ideal, but it helps for now, and as `Counterparty` is a model that is never updates, it has relatively low risk from being inconsistent.